platform/aeternity/api.go

```diff
@@ -0,0 +1,53 @@
+package aeternity
+
+import (
+	"github.com/trustwallet/blockatlas/coin"
+	"net/http"
+
+	"github.com/spf13/viper"
+	"github.com/trustwallet/blockatlas"
+)
+
+type Platform struct {
+	client Client
+}
+
+func (p *Platform) Init() error {
+	p.client.URL = viper.GetString("aeternity.api")
+	p.client.HTTPClient = http.DefaultClient
+	return nil
+}
+
+func (p *Platform) Coin() coin.Coin {
+	return coin.Coins[coin.AE]
+}
+
+func (p *Platform) GetTxsByAddress(address string) (blockatlas.TxPage, error) {
+	addressTxs, err := p.client.GetTxs(address, 25)
+	if err != nil {
+		return nil, err
+	}
+
+	var txs []blockatlas.Tx
+	for _, srcTx := range addressTxs {
+		txs = append(txs, NormalizeTx(&srcTx))
+	}
+	return txs, nil
+}
+
+func NormalizeTx(srcTx *Transaction) blockatlas.Tx {
+	return blockatlas.Tx{
+		ID:     srcTx.Hash,
+		Coin:   coin.AE,
+		From:   srcTx.TxValue.Sender,
+		To:     srcTx.TxValue.Recipient,
+		Fee:    blockatlas.Amount(srcTx.TxValue.Fee),
+		Date:   int64(srcTx.Timestamp) / 1000,
+		Block:  srcTx.BlockHeight,
+		Memo:   "",
```

Comment:
We can try base64 decoding `srcTx.payload` and if it's readable string, we can set it as `Memo`. e.g.: remove `ba_` and last 4 bytes checksum from `ba_SGVsbG8sIE1pbmVyISAvWW91cnMgQmVlcG9vbC4vKXcQag==` and decode, you will get `Hello, Miner! /Yours Beepool./`

Reply:
Can you link me to the docs, seems like the checksum might be bigger than 4 bytes
@hewigovens

Reply:
> Can you link me to the docs, seems like the checksum might be bigger than 4 bytes
> @hewigovens
nvm, I actually find out it's 8 bytes checksum.
https://github.com/aeternity/aepp-sdk-python/pull/204/files

Reply:
I didn't find the docs, here is code shows how to encode the `payload`:

- https://github.com/aeternity/aepp-sdk-js/blob/73552e5aee/es/tx/builder/index.js#L54
- https://github.com/aeternity/aepp-sdk-js/blob/73552e5aee/es/utils/crypto.js#L161